internal/controller/common/helper.go

```diff
@@ -793,3 +794,87 @@ func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
 	}
 	return ud, nil
 }
+
+const (
+	storageClassDefaultAnnotation = "storageclass.kubernetes.io/is-default-class"
+)
+
+func storageClassSupportsVolumeExpansion(ctx context.Context, c client.Client, className *string) (bool, error) {
+	storageClass, err := getStorageClassOrDefault(ctx, c, className)
+	if err != nil {
+		return false, fmt.Errorf("getStorageClassOrDefault failed: %w", err)
+	}
+	return *storageClass.AllowVolumeExpansion, nil
+}
+
+func getStorageClassOrDefault(ctx context.Context, c client.Client, scName *string) (*storagev1.StorageClass, error) {
+	storageClass := &storagev1.StorageClass{}
+	if scName == nil {
+		storageClasses := &storagev1.StorageClassList{}
+		if err := c.List(ctx, storageClasses); err != nil {
+			return nil, err
+		}
+		for _, sc := range storageClasses.Items {
+			if sc.Annotations[storageClassDefaultAnnotation] == "true" {
+				return &sc, nil
+			}
+		}
+		return nil, errors.New("no default storage class found")
+	}
+	if err := c.Get(ctx, types.NamespacedName{Name: *scName}, storageClass); err != nil {
+		return nil, err
+	}
+	return storageClass, nil
+}
+
+// ConfigureStorageParams ...
+type ConfigureStorageParams struct {
+	DesiredSize            resource.Quantity
+	StorageClass           *string
+	DisableVolumeExpansion bool
+	CurrentSize            resource.Quantity
+	SetStorageSizeFunc     func(resource.Quantity)
+}
+
+// ConfigureStorage handles storage configuration and volume expansion checks for the given database cluster.
+func ConfigureStorage(ctx context.Context, c client.Client, db *everestv1alpha1.DatabaseCluster, params ConfigureStorageParams) error {
+	meta.RemoveStatusCondition(&db.Status.Conditions, everestv1alpha1.ConditionTypeCannotExpandStorage)
+
+	hasStorageExpanded := params.CurrentSize.Cmp(params.DesiredSize) < 0 && !params.CurrentSize.IsZero()
```

Comment:
would it make sense to add a condition for the case the user tries to downscale the volume? perhaps ignoring is just fine to not complicate things. just a thought, wanted to bring it to consideration

Reply:
makes sense, updated